File: leetcode/minSubArrayLen.cpp
class Solution {
public:
    int minSubArrayLen(int target, vector<int>& nums) {
	  int left = 0;
	  int total = 0;
      const int init = 1001001001;
	  int res = init;

	  for (int right = 0; right < nums.size(); right++){
		total += nums[right];

		while (target <= total) {
			res = min(res, right - left + 1);
			total -= nums[left++];
		  }
	  }
	  if (res == init) {
		  return 0;
		}
	  return res;
    }
};

File: leetcode/minSubArrayLen.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        tot = nums[0]
        left = 0
        right = 1
        res = M = 1001001001
        if tot >= target:
            res = 1
        while right < len(nums):
            while tot < target and right < len(nums):
                tot += nums[right]
                right += 1
            print(left, right)
            while tot >= target and left < right:
                res = min(res, right-left)
                tot -= nums[left]
                left += 1

        return res if res != M else 0
